### PR TITLE
Bug 1828958: Fix handling of Operand conditions

### PR DESCRIPTION
**Pull request by spadgett**
Check `condition.status` before using a condition as the operand status. Also make sure `phase`, `status`, or `state` are strings before displaying them.

/assign @benjaminapetersen

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx [hunk 1]
```diff
@@ -515,36 +502,124 @@ describe(ProvidedAPIPage.displayName, () => {
   });
 });
 
-describe('OperandStatusIconAndText', () => {
-  let wrapper: ShallowWrapper<OperandStatusIconAndTextProps>;
+describe('OperandStatus', () => {
+  let wrapper: ShallowWrapper<OperandStatusProps>;
 
-  it('dispalys the correct status and status type for a status value of running', () => {
+  it('dispalys the correct status a `status` value of `Running`', () => {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx:508]:
```suggestion
  it('displays the correct status for a `status` value of `Running`', () => {
```

File: frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx [hunk 2]
```diff
@@ -515,36 +502,124 @@ describe(ProvidedAPIPage.displayName, () => {
...
     };
-    wrapper = shallow(<OperandStatusIconAndText statusObject={obj.status} />);
-    expect(wrapper.childAt(0).text()).toEqual(OperatorStatusTypeText[OperatorStatusType.status]);
+    wrapper = shallow(<OperandStatus operand={obj} />);
+    expect(wrapper.childAt(0).text()).toEqual('Status');
     expect(wrapper.childAt(2).props().title).toEqual('Running');
   });
 
-  it('displays the correct status and status type for a phase value of running', () => {
+  it('displays the correct status a `phase` value of `Running`', () => {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx:526]:
```suggestion
  it('displays the correct status for a `phase` value of `Running`', () => {
```

File: frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx [hunk 3]
```diff
@@ -515,36 +502,124 @@ describe(ProvidedAPIPage.displayName, () => {
...
     };
-    wrapper = shallow(<OperandStatusIconAndText statusObject={obj.status} />);
-    expect(wrapper.childAt(0).text()).toEqual(OperatorStatusTypeText[OperatorStatusType.phase]);
+    wrapper = shallow(<OperandStatus operand={obj} />);
+    expect(wrapper.childAt(0).text()).toEqual('Phase');
     expect(wrapper.childAt(2).props().title).toEqual('Running');
   });
 
-  it('displays Unknown for a missing or unknown status object', () => {
+  it('displays the correct status a `phase` value of `Running`', () => {
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operand/index.spec.tsx:545]:
```suggestion
  it('displays the correct status for a `phase` value of `Running`', () => {
```

**Comment by spadgett** [comment]
thanks @rhamilto 👍 

updated

**Pull request merged**